### PR TITLE
docs(cli): update cli README

### PR DESCRIPTION
**Pull request by zoumo**
**What this PR does**
- add configuration section
- remove mamba in readme

**why we need it**

https://github.com/caicloud/nirvana/issues/50

**Special notes for your reviewer**:

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -30,22 +28,22 @@ The flag is compatible with the [GNU extensions to the POSIX recommendations for
 
 ### Commands
 
-You can migrate your cobra commands to mamba easily.
+You can migrate your cobra commands to CLI easily.
 
-the `cli.NewCommand` receives a `*cobra.Command` to build the whole CLI.
+the `cli.NewCommand` receives a `*cobra.Command` to build the command line interface.
```
Comment (ddysher) [inline cli/README.md]: nit: s/the/The, and do not use newline

File: cli/README.md [hunk 2]
```diff
@@ -55,11 +53,11 @@ func main() {
...
 ### Flags
 
-Flag, different from `pflag.Flag`, is an interface containing the following methods in Mamba.
+Flag, different from `pflag.Flag`, is an interface containing the following methods in CLI.
 
-And it is declarative in Mamba, unlike in Cobra where it is imperative which makes the flags **more readable**.
+And it is declarative in CLI, unlike in Cobra where it is imperative, which makes the flags **more readable**.
 
-Mamba will bind `viper` and `flags` automatically. That means you can get all your flag values from `viper`.
+CLI will bind `viper` and `flags` automatically. That means you can get all your flag values from `viper`.
```
Comment (ddysher) [inline cli/README.md]: Do you still need to mention `viper` here?  If so, please link and explain what it is.  (I'd prefer we drop the new concept for users).

File: cli/README.md [hunk 3]
```diff
@@ -55,11 +53,11 @@ func main() {
 
 ### Flags
 
-Flag, different from `pflag.Flag`, is an interface containing the following methods in Mamba.
+Flag, different from `pflag.Flag`, is an interface containing the following methods in CLI.
 
-And it is declarative in Mamba, unlike in Cobra where it is imperative which makes the flags **more readable**.
+And it is declarative in CLI, unlike in Cobra where it is imperative, which makes the flags **more readable**.
```
Comment (ddysher) [inline cli/README.md]: nit: do not add newline

File: cli/README.md [hunk 4]
```diff
@@ -124,21 +122,21 @@ f := cli.StringFlag{
...
 
 -   `AutomaticEnv()`
 -   `SetEnvPrefix(string)`
 -   `SetEnvKeyReplacer(*strings.Replacer)`
 
-*When working with ENV variables, it’s important to recognize that Mamba treats ENV variables case insensitively. All ENV variables are treated as UPPER case*
+*When working with ENV variables, it’s important to recognize that CLI treats ENV variables case insensitively. All ENV variables are treated as UPPER case*
 
-By using `AutomaticEnv`, you tell mamba to bind all flags with ENV automatically. 
+By using `AutomaticEnv`, you tell CLI to bind all flags with ENV automatically.
```
Comment (ddysher) [inline cli/README.md]: nit: this is a little messy, can we use list?

File: cli/README.md [hunk 5]
```diff
@@ -172,18 +170,89 @@ You can set the flag as usual.
...
+#### Persistent
+
+persistent option makes the flag can be inherited by it children‘s commands
+
+### Configuration 
+
+CLI can be also treated as a configuration registry.
 
-You can get a flag value from viper easily.
+The flags are binding to registry automatically. That means all defined flags' values can be accessed by `Get()` function.
```
Comment (ddysher) [inline cli/README.md]: s/binding/bound

File: cli/README.md [hunk 6]
```diff
@@ -172,18 +170,89 @@ You can set the flag as usual.
 -   `Hidden` just hide the flag
 -   `Deprecated` and `ShorthandDeprecated` will print a deprecated message to you if you use the flag
 
-### [WIP] Viper 
+#### Persistent
+
+persistent option makes the flag can be inherited by it children‘s commands
```
Comment (ddysher) [inline cli/README.md]: s/persistent/Persistent

File: cli/README.md [hunk 7]
```diff
@@ -172,18 +170,89 @@ You can set the flag as usual.
...
 ```go
-cli.Viper.GetString("log")
+cmd.AddFlag(cli.StringFlag{Name: "log", DefValue: "test configuration"})
+cli.GetString("log") // test configuration
 ```
 
->   Need further integration
+#### Reading Config Files
+
+There are two way for you to let CLI know where to look for config files.
```
Comment (ddysher) [inline cli/README.md]: s/way/ways

File: cli/README.md [hunk 8]
```diff
@@ -172,18 +170,89 @@ You can set the flag as usual.
...
+-   `GetUint32(key string) uint32`
+-   `GetUint64(key string) uint64`
 
+>   Note that each Get function will return a zero value if it is not found. To check if a given key exists, please use `IsSet()`
 
 ## Thanks
 
-spf13 creates awesome tools to make it easier to build a beautiful modern CLI apps.
\ No newline at end of file
+spf13 creates awesome tools to make it easier to build a beautiful modern CLI apps.
```
Comment (ddysher) [inline cli/README.md]: nit: thanks [spf13](https://github.com/spf13) for creating awesome ...

File: cli/README.md [hunk 9]
```diff
@@ -1,10 +1,8 @@
-# Cli - Mamba
-
-The CLI pkg is also named Mamba.
+# CLI
 
 ## About the project
```
[the remark this answers is not in the document]
Comment (zoumo) [reply cli/README.md]: package descriptions are in `About the project` section

**Pull request merged**